server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestPutMappingAction.java

```diff
@@ -81,19 +81,19 @@ public String getName() {
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
         PutMappingRequest putMappingRequest = putMappingRequest(Strings.splitStringByCommaToArray(request.param("index")));
 
-        boolean includeTypeName = request.paramAsBoolean(INCLUDE_TYPE_NAME_PARAMETER,
-            DEFAULT_INCLUDE_TYPE_NAME_POLICY);
         String type = request.param("type");
         Map<String, Object> sourceAsMap = XContentHelper.convertToMap(request.requiredContent(), false,
             request.getXContentType()).v2();
 
-        if (includeTypeName) {
+        if (request.hasParam(INCLUDE_TYPE_NAME_PARAMETER) == false) {
             deprecationLogger.deprecatedAndMaybeLog("put_mapping_with_types", TYPES_DEPRECATION_MESSAGE);
         } else if (type != null || isMappingSourceTyped(MapperService.SINGLE_MAPPING_NAME, sourceAsMap)) {
```

Comment:
I am set up include_type_name=true, do we still want to allow typed "put mapping request" ?
From what it looks like now, my typed request will fail even if I set include_type_name=true.

Reply:
+1 I am fixing this logic now, it was also caught by our tests.